tools/nebula-spark/src/main/scala/com/vesoft/nebula/tools/connector/reader/NebulaRelation.scala

```diff
@@ -38,40 +47,34 @@ private case class NebulaRelation(override val sqlContext: SQLContext, nebulaOpt
     val metaClient                      = NebulaUtils.createMetaClient(nebulaOptions.getHostAndPorts, nebulaOptions)
 
     import scala.collection.JavaConverters._
-    var schemaColAndType: mutable.Map[String, Class[_]] = mutable.Map[String, Class[_]]()
+    var schemaCols: Seq[ColumnDef] = Seq()
 
     returnColMap.keySet.foreach(k => {
-      if (DataTypeEnum.VERTEX.toString.equalsIgnoreCase(nebulaOptions.dataType)) {
+      val schema = if (DataTypeEnum.VERTEX.toString.equalsIgnoreCase(nebulaOptions.dataType)) {
         fields.append(DataTypes.createStructField("_vertexId", DataTypes.StringType, false))
-        schemaColAndType =
-          metaClient.getTagSchema(nebulaOptions.spaceName, nebulaOptions.label).asScala
+        metaClient.getTag(nebulaOptions.spaceName, nebulaOptions.label)
       } else {
         fields.append(DataTypes.createStructField("_srcId", DataTypes.StringType, false))
         fields.append(DataTypes.createStructField("_dstId", DataTypes.StringType, false))
-        schemaColAndType =
-          metaClient.getEdgeSchema(nebulaOptions.spaceName, nebulaOptions.label).asScala
+        metaClient.getEdge(nebulaOptions.spaceName, nebulaOptions.label)
       }
+      schemaCols = schema.columns.asScala
       if (nebulaOptions.allCols) {
         // if allCols is true, then fields should contain all properties.
-        schemaColAndType.keySet
-          .foreach(propName => {
-            LOG.info(s"prop name ${propName}, type ${schemaColAndType} ")
-            fields.append(
-              DataTypes.createStructField(propName,
-                                          NebulaUtils.convertDataType(schemaColAndType(propName)),
-                                          true))
+        schemaCols
+          .foreach(columnDef => {
+            LOG.info(s"prop name ${columnDef.getName}, type ${columnDef.getType}")
+            fields.append(DataTypes.createStructField(columnDef.getName, columnDef2dataType(columnDef), true))
```

Comment:
GREAT WORK! 
We should know that spark reader reads the latest version of tag and edge.